### ISSUE TITLE
a11y-no-noninteractive-tabindex should recognize role="application" as interactive

### ISSUE THREAD

**Issue by chbert** [pheuter/svelte-check-rs#49]
## Description

The `a11y-no-noninteractive-tabindex` warning incorrectly flags elements with `tabindex="0"` when they have `role="application"`, which makes them semantically interactive.

According to [WAI-ARIA](https://www.w3.org/TR/wai-aria-1.1/#application), `role="application"` declares a region as a web application rather than a web document, making it an interactive widget that should be focusable.

## Minimal Reproduction

```svelte
<!-- This should NOT produce a warning -->
<div
  tabindex="0"
  role="application"
  aria-label="Interactive canvas"
>
  <canvas></canvas>
</div>
```

## Current Behavior

```
Warning: A11y: Non-interactive elements like <div> should not have tabindex (a11y-no-noninteractive-tabindex)
```

## Expected Behavior

No warning, because `role="application"` makes the element interactive.

## Additional Context

Other interactive roles that should also be recognized include:
- `role="button"`
- `role="link"`
- `role="menuitem"`
- `role="tab"`
- `role="slider"`
- `role="textbox"`
- etc.

The official `svelte-check` handles this correctly by not warning when an interactive role is present.

## Environment

- svelte-check-rs version: latest
- OS: macOS
